tests/timerun_shared_tests.cpp

```diff
@@ -5,54 +5,75 @@ using namespace ETJump;
+    return {-1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, -1, -1};
+  }
 };
 
 TEST_F(TimerunSharedTests, Start_ShouldSerialize) {
-  auto start = TimerunCommands::Start(1, 2, "3", 4, CreateDefaultCheckpoints());
+  auto start = TimerunCommands::Start(1, 2, "3", 4, CreateDefaultCheckpoints(),
+                                      CreateDefaultCurrentCheckpoints());
 
-  ASSERT_EQ(start.serialize(),
-            "timerun start 1 2 \"3\" 4 \"1,2,3,4,5,6,7,8,9,10,11,12,13,14,-1,-1\"");
+  ASSERT_EQ(
+      start.serialize(),
+      "timerun start 1 2 \"3\" 4 \"1,2,3,4,5,6,7,8,9,10,11,12,13,14,-1,-1\" "
+      "\"-1,2,3,4,5,6,7,8,9,10,11,12,13,14,-1,-1\"");
 }
 
 TEST_F(TimerunSharedTests, Start_ShouldDeserialize) {
   auto args =
-      std::vector<std::string>{"timerun", "start", "1", "2", "3", "4",
-                               "1,2,3,4,5,6,7,8,9,10,11,12,13,14,-1,-1"};
+      std::vector<std::string>{"timerun",
+                               "start",
+                               "1",
+                               "2",
+                               "3",
+                               "4",
+                               "1,2,3,4,5,6,7,8,9,10,11,12,13,14,-1,-1",
+                               "-1,2,3,4,5,6,7,8,9,10,11,12,13,14,-1,-1"};
   auto deserialized = TimerunCommands::Start::deserialize(args);
 
   ASSERT_EQ(deserialized.value().clientNum, 1);
   ASSERT_EQ(deserialized.value().startTime, 2);
   ASSERT_EQ(deserialized.value().runName, "3");
   ASSERT_EQ(deserialized.value().previousRecord.value(), 4);
   ASSERT_EQ(deserialized.value().checkpoints[10], 11);
+  ASSERT_EQ(deserialized.value().currentRunCheckpoints[0], -1);
 }
 
 TEST_F(TimerunSharedTests, Start_ShouldSerialize_IfNoPreviousRecord) {
-  auto start = TimerunCommands::Start(1, 2, "3", opt<int>(),
-                                      CreateDefaultCheckpoints());
-
-  ASSERT_EQ(start.serialize(),
-            "timerun start 1 2 \"3\" -1 \"1,2,3,4,5,6,7,8,9,10,11,12,13,14,-1,-1\"");
+  auto start =
+      TimerunCommands::Start(1, 2, "3", opt<int>(), CreateDefaultCheckpoints(),
+                             CreateDefaultCurrentCheckpoints());
+
+  ASSERT_EQ(
+      start.serialize(),
+      "timerun start 1 2 \"3\" -1 \"1,2,3,4,5,6,7,8,9,10,11,12,13,14,-1,-1\" "
+      "\"-1,2,3,4,5,6,7,8,9,10,11,12,13,14,-1,-1\"");
 }
 
 TEST_F(TimerunSharedTests, Start_ShouldDeserialize_IfNoPreviousRecord) {
-  auto args = std::vector<std::string>{"timerun", "start", "1", "2", "3", "-1",
-                                       "1,2,3,4,5,6,7,8,9,10,11,12,13,14,-1,-1"};
+  auto args =
+      std::vector<std::string>{"timerun",
+                               "start",
+                               "1",
+                               "2",
+                               "3",
+                               "-1",
+                               "1,2,3,4,5,6,7,8,9,10,11,12,13,14,-1,-1",
+                               "-1,2,3,4,5,6,7,8,9,10,11,12,13,14,-1,-1"};
```

Comment:
guess it'd make sense to have a valid test case (e.g. first element should be >= 0) but i dunno